### ISSUE TITLE
Accuracy Decreases Tremendously after IR conversion of model 

### ISSUE THREAD

**Issue by divdaisymuffin** [OpenVisualCloud/Smart-City-Sample#792]
Hi @nnshah1 and @xwu2git,

Need Your help with, We have trained a head detection model on yolov3 and tested it on opencv, Where we have recieved good accuracy.
Video is attached:
https://user-images.githubusercontent.com/68895288/136757909-cf913224-6b02-4e0f-adee-b3f95b06c33a.mp4

Later we have converted the Model to IR using provided documents via intel [Conversion](https://github.com/PINTO0309/OpenVINO-YoloV3/wiki/Reference-repository#conversion-success-2)
When we tested on same video accuracy decreased drastically. Please see the video and model proc.
Video:
https://user-images.githubusercontent.com/68895288/136762446-27935935-1a31-4d78-9be0-6f45451dba97.mp4

modelproc:
 
`{
  "json_schema_version": "2.0.0",
  "input_preproc": [],
  "output_postproc": [
    {
      "converter": "tensor_to_bbox_yolo_v3",
      "iou_threshold": 0.5,
      "classes": 1,
      "anchors": [
        10.0,
        13.0,
        16.0,
        30.0,
        33.0,
        23.0,
        30.0,
        61.0,
        62.0,
        45.0,
        59.0,
        119.0,
        116.0,
        90.0,
        156.0,
        198.0,
        373.0,
        326.0
      ],
      "masks": [
        6,
        7,
        8,
        3,
        4,
        5,
        0,
        1,
        2
      ],
      "bbox_number_on_cell": 3,
      "cells_number": 13,
      "labels": ["person"]
    }
  ]
}
`

We tried another approach,where we have converted the Yolov3 to TF separately and then frozen the model and converted to IR format.
But surprisingly, we are not able to get detection, model proc you can check.
Model Proc:
`Proc File: 
{
  "json_schema_version": "2.0.0",
  "input_preproc": [],
  "output_postproc": [
    {
      "labels": ["person"],
      "converter": "tensor_to_bbox_yolo_v3"
    }
  ]
}
`

So we want to know, Why there is so drastic decrease in accuracy? Did we missed something while conversion and in second approach where we have converted yolo to tf externally, why we are getting no detection? is model proc is a problem?

**Comment by divdaisymuffin** [comment]
@nnshah1 , Thanks,  We will get back to you with all information

**Comment by divdaisymuffin** [comment]
Hi @nnshah1 , We have tried your suggested method as well, but still accuracy is degraded only.
I am providing you the link where you can find the IR converted model, and original model file.

https://drive.google.com/drive/folders/1_nHqMxvlxuFhpJBEma3uKZPBo6imKqXl

**Comment by vidyasiv** [comment]
Hi @divdaisymuffin, thanks for providing the model information, could you share the original video used i.e without watermarking?

**Comment by divdaisymuffin** [comment]
@nnshah1 yes after changing the model proc, got us better results on Vaserving.

**Comment by divdaisymuffin** [comment]
@nnshah1 @vidyasiv Thanks a lot for the prompt support.

**Comment by nnshah1** [comment]
@divdaisymuffin Thanks for confirming! 

In addition @vidyasiv spent some time following the conversion steps (as recommended to avoid the change to model proc) and was successful using the following resources and notes and achieve the expected accuracy and confirmed that **reverse_input_channels** was set.

Steps outlined here: 

[https://docs.openvino.ai/latest/openvino_docs_MO_DG_prepare_model_convert_model_tf_specific_Convert_YOLO_From_Tensorflow.html#convert-yolov3-model-to-ir](https://docs.openvino.ai/latest/openvino_docs_MO_DG_prepare_model_convert_model_tf_specific_Convert_YOLO_From_Tensorflow.html#convert-yolov3-model-to-ir)

Notes on steps:

- For Dump YOLOv3 TensorFlow* Model
                - Used Python virtual environment to get Python3.5 version, pip install tensorflow==1.11 for repository https://github.com/mystic123/tensorflow-yolo-v3
                - Within virtual env, ran with Original model names and weights 
                                 python3 convert_weights_pb.py --class_names classes.names --data_format NHWC --weights_file yolov3_custom_train_20000.weights
                - Output: TF_Model/frozen_darknet_yolov3_model.pb

- For Convert YOLOv3 TensorFlow Model to IR
                - Openvino: 2021.4.689 (datadev image)
                - Use a modified version of yolo_v3.json to have classes: 1
                - Run
                                python3 deployment_tools/model_optimizer/mo_tf.py --input_model /home/video-analytics-serving/models/TF_Model/frozen_darknet_yolov3_model.pb --transformations_config /home/video-analytics-serving/models/Converted_Model_TF/yolo_v3.json --batch 1  --output_dir /home/video-analytics-serving/models/Converted_Model_TF --reverse_input_channels
                (I also tried mo.py and results seemed identical)

                The output printed clarifies the setting of reverse_input_channels.
                Model Optimizer arguments:
                Common parameters:
        - Path to the Input Model:      /home/video-analytics-serving/models/TF_Model/frozen_darknet_yolov3_model.pb
        - Path for generated IR:        /home/video-analytics-serving/models/Converted_Model_TF
        - IR output name:       frozen_darknet_yolov3_model
        - Log level:    ERROR
        - Batch:        1
        - Input layers:         Not specified, inherited from the model
        - Output layers:        Not specified, inherited from the model
        - Input shapes:         Not specified, inherited from the model
        - Mean values:  Not specified
        - Scale values:         Not specified
        - Scale factor:         Not specified
        - Precision of IR:      FP32
        - Enable fusing:        True
        - Enable grouped convolutions fusing:   True
        - Move mean values to preprocess section:       None
        - Reverse input channels:       True
                TensorFlow specific parameters:
        - Input model in text protobuf format:  False
        - Path to model dump for TensorBoard:   None
        - List of shared libraries with TensorFlow custom layers implementation:        None
        - Update the configuration file with input/output node names:   None
        - Use configuration file used to generate the model with Object Detection API:  None
        - Use the config file:  None
        - Inference Engine found in:    /opt/intel/openvino_2021.4.689/python/python3.8/openvino
Inference Engine version:       2021.4.1-3926-14e67d86634-releases/2021/4
Model Optimizer version:        2021.4.1-3926-14e67d86634-releases/2021/4
[ SUCCESS ] Generated IR version 10 model.
[ SUCCESS ] XML file: /home/video-analytics-serving/models/Converted_Model_TF/frozen_darknet_yolov3_model.xml
[ SUCCESS ] BIN file: /home/video-analytics-serving/models/Converted_Model_TF/frozen_darknet_yolov3_model.bin
[ SUCCESS ] Total execution time: 41.92 seconds.
[ SUCCESS ] Memory consumed: 1721 MB.
